crates/sui-core/src/authority.rs

```diff
@@ -851,10 +851,19 @@ impl AuthorityState {
                 } else {
                     error!(?digest, "commit_certificate failed: {}", err);
                 }
-                debug!("Failed to notify ticket with sequence number: {}", seq);
+                debug!(
+                    "Failed to notify ticket with sequence number: {}",
+                    ticket_seq
+                );
                 return Err(err);
             }
-            Ok(seq) => seq,
+            Ok(seq) => {
+                if seq < ticket_seq {
+                    debug!("Notifying during retry, current low watermark {:?}, ticket_seq {:?}, seq {:?}", self.batch_notifier.low_watermark(), ticket_seq, seq);
+                    self.batch_notifier.notify(seq);
```

Comment:
the `notifier_ticket.notify();` in `fn commit_certificate` does not work, is it because it only notifies the `ticket_seq` but not the old one (`seq`) ?
Is it better to move the notify-old-seq logic to the end of `fn commit_certificate` where `notifier_ticket.notify();` happens?

https://github.com/MystenLabs/sui/blob/qd-retry-locked-objs/crates/sui-core/src/authority.rs#L2127

Reply:
@longbowlu lmk if you strongly feel about moving this check in `commit_certificate` but i like the fact that all tricky batch notifier checks are in one place here

Reply:
no strong opinions